tools/codegen/generate.bzl

```diff
@@ -110,4 +110,41 @@ def mixer_proto_library(
       name = name,
       srcs = [name + "_handler.gen.go"],
       library = ":" + name + "_gogo_proto",
-      deps = deps + MIXER_DEPS)
\ No newline at end of file
+      deps = deps + MIXER_DEPS)
+
+###############
+
+def _mixer_supported_template_gen(name, packages, out):
+  args = ""
+  descriptors = []
+  for k1, v in packages.items():
+    l = "$(location %s)" % (k1)
+    args += " %s:%s " % (l, v)
+    descriptors.append(k1)
+
+  native.genrule(
+      name = name+"_gen",
+      outs = [out],
+      cmd = "$(location //tools/codegen/cmd/mixgenbootstrap) " + args + " -o $(location %s)" % (out),
+      tools = ["//tools/codegen/cmd/mixgenbootstrap"] + descriptors,
```

Comment:
why are the descriptors in the tools section?  What does that accomplish? Should they be in `srcs` instead?

I would only expect `mixgenbootstrap` to be in `tools`, as that is the cmd being run.

Reply:
descriptors are there in the tools, in order to make the [line 121 " l = "$(location %s)" % (k1)"](https://github.com/istio/mixer/pull/1199/files#diff-c8bd90f421d92c7dfb72908cbd50081cR121)  work . I guess they are sources to be precise..  Made the change